Python/compile.c

```diff
@@ -1839,6 +1840,23 @@ compiler_add_yield_from(struct compiler *c)
     return 1;
 }
 
+static int
+compiler_pop_except_and_reraise(struct compiler *c)
+{
+    /* Stack contents
+     * [exc_info, lasti, exc]  ROT_THREE
+     * [exc, exc_info, lasti]  ROT_THREE
```

Comment:
We'd use a reverse `ROT_THREE` if we had one, huh? 🙂  Seriously though, it feels like there should be something the compiler could to make these two instructions unnecessary.  Manually reordering the stack like this feels like some sort of shim.  That said, clearly we use `ROT_*` in various places.  Is that just how we adapt the stack between different opcodes?

Reply:
In this case the exc_info was pushed before we entered the nested block and then the exception was pushed by some raise inside the block, so something needs to reorder them.

Reply:
It makes sense when you put it that way. 🙂

Reply:
Actually after our discussion yesterday I realised we can use COPY(3) instead of the two rotations.  I just pushed that.

It copies the exc_info to TOS for POP_EXCEPT, and leaves the original at the bottom of the stack which gets cleared in exception_unwind.